src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs

```diff
@@ -575,8 +608,13 @@ public void AnalyzeCommand_CachesResultsWhenComputingTargetHashesAndPersistingTo
             RunTests(persistLogFileToDisk: true);
         }
 
-        private static void RunTests(bool persistLogFileToDisk)
+        private static void RunTests(bool persistLogFileToDisk, bool raiseNotificationPerBinary = false, int expectedReturnCode = 0)
         {
+            // TODO: we require a special case here currently because in cases where we are not persisting to a log file and we
+            // are raising an exception per binary, these notifications are converted to results by the MSBuildConverter. In 
+            // future, we should fix the converter to 
```

Comment:
>to  [](start = 51, length = 3)

"... generate tool notifications rather than results for console output lines of the form '<toolName> : ...'"  ?